### ISSUE TITLE
[Bug] Process hangs on certain trial

### ISSUE THREAD

**Issue by nixonthe** [ray-project/ray#21418]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Tune

### What happened + What you expected to happen

I'm trying to tune my DL model and my tuning process hangs on certain trial - 16. I've tried to rerun experiment and this issue repeats. On different machine issue doesn't show. How to solve it?

== Status ==
Current time: 2022-01-06 11:14:13 (running for 00:11:36.74)
Memory usage on this node: 21.1/63.8 GiB
Using FIFO scheduling algorithm.
Resources requested: 1.0/16 CPUs, 1.0/1 GPUs, 0.0/27.47 GiB heap, 0.0/13.74 GiB objects
Current best trial: 648d3_00000 with val_loss=0.5780708092334738 and parameters={'hidden_size': [22259, 5987, 15957], 'early_stopping': 5, 'drop_rate': 0.06286198471735105, 'L2': 0.023992738384729536, 'batch_size': 512, 'learning_rate': 2.069069999940908e-05}
Result logdir: D:\users\glazunov\torch_pipelines\ray_reports\05_01_x2
Number of trials: 33/5000 (17 PENDING, 1 RUNNING, 15 TERMINATED)

As you can see there is no error but trial just hangs.

### Versions / Dependencies

python = 3.9
ray = 1.9.0
Windows Server 2019 Standart

### Reproduction script

def train(model, train_loader, criterion, optimizer, device):
    train_loss = 0
    model.train()
    for i, (embeds, labels) in enumerate(train_loader):
        embeds = embeds.to(device).type(torch.float32)
        labels = labels.to(device)

        outputs = model(embeds)
        loss = criterion(outputs, labels)
        train_loss += loss.item() * len(labels)

        optimizer.zero_grad()
        loss.backward()
        optimizer.step()

    return train_loss

def validate(model, validate_loader, criterion, device):
    model.eval()
    with torch.no_grad():
        val_loss = 0
        val_acc = 0
        for i, (embeds, labels) in enumerate(validate_loader):
            # Move tensors to the configured device
            embeds = embeds.to(device).type(torch.float32)
            labels = labels.to(device)
            outputs = model(embeds)
            y_preds = model.softmax(outputs).argmax(dim=1)
            val_loss += criterion(outputs, labels).item() * len(labels)
            val_acc += (y_preds == labels.argmax(dim=1)).cpu().numpy().mean().item() * len(labels)

    return val_loss, val_acc

def train_with_raytune(config, data=None):
    torch.manual_seed(18)
    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
    train_set = data["train_set"]
    validate_set = data["validate_set"]
    columns_to_drop = data["columns_to_drop"]
    class_labels = data["class_labels"]
    class_matcher = {label: idx for idx, label in enumerate(class_labels)}
    train_dataset = CustomDataset(train_set.drop(columns=columns_to_drop).values,
                                  onehot_matcher(np.vectorize(class_matcher.get)(train_set.y_real.values)))
    validate_dataset = CustomDataset(validate_set.drop(columns=columns_to_drop).values,
                                     onehot_matcher(np.vectorize(class_matcher.get)(validate_set.y_real.values)))
    input_size = train_set.shape[1] - len(columns_to_drop)
    early_stopping = config["early_stopping"]
    drop_rate = config["drop_rate"]
    L2 = config["L2"]
    batch_size = config["batch_size"]
    learning_rate = config["learning_rate"]
    output_size = len(class_labels)
    hidden_size = config['hidden_size']
    class_weight = None

    model = MLP_softmax(input_size, hidden_size, output_size, drop_rate).to(device)
    criterion = nn.CrossEntropyLoss(weight=class_weight)
    optimizer = torch.optim.Adam(model.parameters(), lr=learning_rate, weight_decay=L2)

    train_loader = torch.utils.data.DataLoader(dataset=train_dataset,
                                               batch_size=batch_size,
                                               shuffle=True)

    validate_loader = torch.utils.data.DataLoader(dataset=validate_dataset,
                                                  batch_size=batch_size,
                                                  shuffle=False)

    val_losses = []
    while True:
        train_loss = train(model, train_loader, criterion, optimizer, device)
        val_loss, val_acc = validate(model, validate_loader, criterion, device)
        val_loss /= len(validate_dataset)
        val_acc /= len(validate_dataset)
        train_loss /= len(train_dataset)
        val_losses.append(val_loss)
        val_loss_no_improvement_step = len(val_losses[-early_stopping:]) - np.argmin(val_losses[-early_stopping:])
        tune.report(train_loss=train_loss, val_loss=val_loss, val_acc=val_acc,
                    val_loss_no_improvement_step=val_loss_no_improvement_step)

analysis = tune.run(
    tune.with_parameters(train_with_raytune, data=data),
    metric="val_loss",
    mode="min",
    name="05_01_x2",
    stop={
        "val_loss_no_improvement_step": 5,
        "training_iteration": 6
    },
    resources_per_trial={
        "gpu": 1
    },
    num_samples=5000,
    config=config,
    verbose=1,
    local_dir="ray_reports"
)

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by nixonthe** [comment]
In tune.run I write reuse_actors=True and it solves the issue.
